File: jwt/src/main/java/com/ufoscout/coreutils/jwt/CoreJsonProvider.java
package com.ufoscout.coreutils.jwt;

import com.ufoscout.coreutils.json.JsonSerializerService;

public class CoreJsonProvider implements JsonProvider {

    private JsonSerializerService jjService;

    public CoreJsonProvider(JsonSerializerService jjService) {
        this.jjService = jjService;
    }

    @Override
    public <T> String toJson(T payload) {
        return jjService.toJson(payload);
    }

    @Override
    public <T> T fromJson(Class<T> payloadClass, String json) {
        return jjService.fromJson(payloadClass, json);
    }
}

File: jwt-kotlin/src/main/java/com/ufoscout/coreutils/jwt/kotlin/CoreJsonProvider.kt
package com.ufoscout.coreutils.jwt.kotlin

import com.ufoscout.coreutils.json.kotlin.JsonSerializerService
import com.ufoscout.coreutils.jwt.JsonProvider

class CoreJsonProvider(private val jjService: JsonSerializerService) : JsonProvider {

    override fun <T : Any> toJson(payload: T): String {
        return jjService.toJson(payload)
    }

    override fun <T : Any> fromJson(payloadClass: Class<T>, json: String): T {
        return jjService.fromJson(payloadClass.kotlin, json)
    }
}
